test/e2e/letters-helpers.js

```diff
@@ -0,0 +1,157 @@
+    }
+  }
+};
+
+const benefitSummaryOptions = {
+  data: {
+    attributes: {
+      benefitInformation: {
+        awardEffectiveDate: '1965-01-01T05:00:00.000+00:00',
+        hasAdaptedHousing: false,
+        hasChapter35Eligibility: true,
+        hasIndividualUnemployabilityGranted: false,
+        hasNonServiceConnectedPension: false,
+        hasServiceConnectedDisabilities: true,
+        hasSpecialMonthlyCompensation: false,
+        monthlyAwardAmount: 123,
+        serviceConnectedPercentage: 2
+      },
+      militaryService: [
+        {
+          branch: 'ARMY',
+          characterOfService: 'HONORABLE',
+          enteredDate: '1965-01-01T05:00:00.000+00:00',
+          releasedDate: '1972-10-01T04:00:00.000+00:00'
+        },
+        {
+          branch: 'ARMY',
+          characterOfService: 'UNCHARACTERIZED_ENTRY_LEVEL',
+          enteredDate: '1973-01-01T05:00:00.000+00:00',
+          releasedDate: '1977-10-01T04:00:00.000+00:00'
+        }
+      ]
+    }
+  }
+};
+
+const address = {
+  data: {
+    attributes: {
+      address: {
+        addressEffectiveDate: '2012-04-03T04:00:00.000+00:00',
+        addressOne: '57 COLUMBUS STRASSA',
+        addressThree: '',
+        addressTwo: 'BEN FRANKLIN VILLAGE',
+        militaryPostOfficeTypeCode: 'APO',
+        militaryStateCode: 'AE',
+        type: 'MILITARY',
+        zipCode: '09028',
+        zipSuffix: ''
+      },
+      controlInformation: {
+        canUpdate: true,
+        corpAvailIndicator: true,
+        corpRecFoundIndicator: true,
+        hasNoBdnPaymentsIndicator: true,
+        indentityIndicator: true,
+        indexIndicator: true,
+        isCompetentIndicator: true,
+        noFiduciaryAssignedIndicator: true,
+        notDeceasedIndicator: true
+      }
+    }
+  }
+};
+
+const newAddress = {
+  addressEffectiveDate: '2012-04-03T04:00:00.000+00:00',
+  addressOne: '57 COLUMBUS STRASSA',
+  addressThree: '',
+  addressTwo: 'BEN FRANKLIN VILLAGE',
+  city: 'Chicago',
+  state: 'IL',
+  type: 'DOMESTIC',
+  zipCode: '60602',
+  zipSuffix: ''
+};
+
+const countries = {
+  data: {
+    attributes: {
+      countries: [
+        'Denmark', 'Sri Lanka', 'USA', 'United Kingdom'
+      ]
+    }
+  }
+};
+
+const states = {
+  data: {
+    attributes: {
+      states: [
+        'IL', 'MA', 'DC'
+      ]
+    }
+  }
+};
+
+// Create API routes
+function initApplicationMock(token) {
+  mock(token, {
+    path: '/v0/letters',
+    verb: 'get',
+    value: letters
+  });
+
+  mock(token, {
+    path: '/v0/letters/beneficiary',
+    verb: 'get',
+    value: benefitSummaryOptions
+  });
+
+  mock(token, {
+    path: '/v0/address',
+    verb: 'get',
+    value: address
+  });
+
+  mock(token, {
+    path: '/v0/address/countries',
+    verb: 'get',
+    value: countries
+  });
+
+  mock(token, {
+    path: '/v0/address/states',
+    verb: 'get',
+    value: states
+  });
+
+  mock(token, {
+    path: '/v0/address',
+    verb: 'put',
+    value: newAddress,
```

Comment:
This mocks returning the `newAddress` when we do a put request, right? Do we know this is what the api returns?

Reply:
I'm not totally sure how this works, but I think there is a special `/mock` endpoint that we're using for these mock requests. I would have assumed `newAddress` in this case is what is being sent in the request body, but I'm not positive how `/mock` works.

Reply:
You're right, this means `newAddress` will be sent in the response when a put request is made. In building the `newAddress` data in the e2e helpers for letters, I based it off of what the real vets-api endpoint returns.